src/NorthwindWebsite/Middleware/ImageCachingMiddleware.cs

```diff
@@ -0,0 +1,110 @@
+﻿using NorthwindWebsite.Core.ApplicationSettings;
+using NorthwindWebsite.Core.Constants;
+using NorthwindWebsite.Core.Utils;
+using NorthwindWebsite.Middleware.Handlers.Interfaces;
+
+namespace NorthwindWebsite.Middleware;
+
+public class ImageCachingMiddleware : IMiddleware
+{
+    private readonly AppSettings _appSettings;
+    private readonly IImageCachingHandler _imageCachingService;
+    private readonly ILogger<ImageCachingMiddleware> _logger;
+
+    public ImageCachingMiddleware(
+        AppSettings appSettings,
+        IImageCachingHandler cacheService,
+        ILogger<ImageCachingMiddleware> logger)
+    {
+        _appSettings = appSettings;
+        _imageCachingService = cacheService;
+        _logger = logger;
+    }
+
+    public async Task InvokeAsync(HttpContext context, RequestDelegate next)
+    {
+        var expiresParsed = DateTime.TryParse(context.Response.Headers["expires"], out var expires);
+
+        var requestPath = context.Request.Path.Value;
+
+        string imageIndex = requestPath!.GetImageIndexFromRequest();
+
+        _imageCachingService.CreateFolderIfDoesNotExists();
+
+        if (_imageCachingService.IsContained(imageIndex))
+        {
+            var indexParsed = int.TryParse(imageIndex, out var index);
```

Comment:
1. For boolean variables try to add where it's possible `is` suffix, it simplifies code reading and bring clear purpose of variable usage and type.
2. `index` is a little bit abstract, let's update it to `parsedImageIndex`